src/core/adapters/CodeGouvApiSemiMock.ts

```diff
@@ -0,0 +1,97 @@
+import type {CodeGouvApi, Dependency, Organisation, Repository, RepositoryStatistics} from "core/ports/CodeGouvApi";
+import axios from "axios";
+import memoize from "memoizee";
+
+export function createCodeGouvApiSemiMock(): CodeGouvApi {
+
+    const url = 'https://code.gouv.fr/data';
+
+    const getRepositories = memoize(async () => {
+        const response = await axios.get(`${url}/repos.json`);
+        return response.data.map((repository: any): Repository => {
+            return {
+                description: repository.d || '',
+                is_experimental: repository.a || false,
+                language: repository.l || '',
+                last_updated: repository.u || 0,
+                latest_tag: '',
+                license: repository.li || '',
+                name: repository.n || '',
+                organisation_name: repository.o || '',
+                sill_id: 0,
+                star_count: repository.s || 0,
+                status: 'Stable',
+                topics: [],
+                type: repository.l ? 'Library' : 'Source Code',
+                url: repository.r,
+                vitality: 100,
```

Comment:
à quoi correspond cette clé "vitality" ?

Reply:
Je veux dire concrètement, à quoi correspond cette indice de vitalité ? Un indice de "propreté du code", un indice d'activité du repo ?

Reply:
Ahh pardon :) oui c'est un indice de qualité qui sera calculé en fonction de plusieurs critères, qui devrait correspondre à ce que l'Italie calcule déjà pour son propre code.gouv : https://github.com/italia/publiccode-crawler/blob/main/vitality-ranges.yml (https://developers.italia.it/en/search?type=software_reuse)